src/helpers/format.ts

```diff
@@ -1,10 +1,53 @@
 import type { ReactTestRendererJSON } from 'react-test-renderer';
-import prettyFormat, { plugins } from 'pretty-format';
+import prettyFormat, { NewPlugin, plugins } from 'pretty-format';
 
-const format = (input: ReactTestRendererJSON | ReactTestRendererJSON[]) =>
+type MapPropsFunction = (
+  props: Record<string, unknown>,
+  node: ReactTestRendererJSON
+) => Record<string, unknown>;
+
+export type FormatOptions = {
+  mapProps?: MapPropsFunction;
+};
+
+const format = (
+  input: ReactTestRendererJSON | ReactTestRendererJSON[],
+  options: FormatOptions = {}
+) =>
   prettyFormat(input, {
-    plugins: [plugins.ReactTestComponent, plugins.ReactElement],
+    plugins: [getCustomPlugin(options.mapProps)],
```

Comment:
I think it's still a good idea to make `getCustomPlugin` tweak only `ReactTestComponent` while passing `ReactElement` normally
```suggestion
    plugins: [getTestComponentPlugin(options.mapProps), plugins.ReactElement],
```

Reply:
Absolutely, it also makes the code way simpler !